src/Bundle/DependencyInjection/SyliusGridExtension.php

```diff
@@ -60,8 +62,23 @@ public function load(array $configs, ContainerBuilder $container): void
             ->addTag('sylius.grid')
         ;
 
+        $container->registerAttributeForAutoconfiguration(
+            AsFilter::class,
+            static function (ChildDefinition $definition, AsFilter $attribute, \Reflector $reflector): void {
+                // Helps to avoid issues with psalm
+                if (!$reflector instanceof \ReflectionClass) {
+                    return;
+                }
+
+                $definition->addTag(AsFilter::SERVICE_TAG, [
+                    'type' => $attribute->type ?? $reflector->getName(),
+                    'form_type' => $attribute->formType,
+                ]);
+            },
+        );
+
         $container->registerForAutoconfiguration(FilterInterface::class)
```

Comment:
we should use this interface instead, that was a mistake to use it cause the FilterInterface itself is not enough to auconfigure it.

```suggestion
        $container->registerForAutoconfiguration(ConfigurableFilterInterface::class)
```